### PR TITLE
Sanitize MSVC builds and enable compiler-warnings job in CI

### PR DESCRIPTION
**Pull request by nurmukhametov**
This change enforces `c++17` standard for all projects including `check_isa.cpp`. It also specifies `/EHsc` for MSVC builds to properly enable exception handling. It fixes macro catching invocation in `ispcrt.cpp` and some minor warnings.

This change makes [compiler-warnings](https://github.com/ispc/ispc/actions/workflows/compiler-warnings.yml) green and enables it for pull requests.

### CODE REVIEW

File: ispcrt/ispcrt.cpp [hunk 1]
```diff
@@ -182,23 +182,23 @@ void ISPCLaunch(void **handlePtr, void *f, void *data, int countx, int county, i
...
+    abort();
 }
 
 void ISPCSync(void *handle) {
     if (ispc_sync_fptr) {
         return ispc_sync_fptr(handle);
     }
     // TODO: this code can be called directly from ISPC code. Not sure how to exit error safely.
-    throw std::runtime_error("Missing ISPCSync symbol");
+    abort();
```
Comment (e3m3) [inline ispcrt/ispcrt.cpp:201]: Should we write the missing symbol message that was removed to stderr before aborting?

Comment (nurmukhametov) [reply ispcrt/ispcrt.cpp]: Yes, we should.

**Review by e3m3 — APPROVED** [review]

File: alloy.py [hunk 2]
```diff
@@ -340,18 +340,20 @@ def unsupported_llvm_targets(LLVM_VERSION):
...
-        if options.ispc_build_compiler == "clang":
-            cisa_compiler = "clang"
-        elif options.ispc_build_compiler == "gcc":
-            cisa_compiler = "g++"
 
-        try_do_LLVM("build check_ISA", cisa_compiler + " check_isa.cpp -o check_isa.exe", True)
-    else:
-        try_do_LLVM("build check_ISA", "cl check_isa.cpp", True)
+    cmd_conf = "cmake . -B check_isa_build -DISPC_INCLUDE_UTILS_ONLY=ON"
+    cmd_build = "cmake --build check_isa_build --target check_isa"
```
Comment (aneshlya) [inline alloy.py:349]: Minor note: I think CMAKE_BUILD_TYPE is Release by default but `cmake --build` builds Debug by default. Let's specify configuration explicitly: 
```suggestion
    cmd_build = "cmake --build check_isa_build --target check_isa --config Release"
```

File: alloy.py [hunk 3]
```diff
@@ -340,18 +340,20 @@ def unsupported_llvm_targets(LLVM_VERSION):
...
+    check_isa = "check_isa"
+    if current_OS == "Windows":
+        check_isa = join("Debug", "check_isa.exe")
```
Comment (aneshlya) [inline alloy.py:352]:
```suggestion
        check_isa = join("Release", "check_isa.exe")
```

File: alloy.py (hunk 2 above)
Comment (e3m3) [reply alloy.py:349]: > but `cmake --build` builds Debug by default

Interesting, I didn't know this. Is using `--config Release` in the build line different from specifying `-DCMAKE_BUILD_TYPE=Release` in the `cmd_conf` line?

Comment (aneshlya) [reply alloy.py:349]: I think it is. Our default build type (Release) for CMake configuration is set in CMakeLists.txt: https://github.com/ispc/ispc/blob/main/CMakeLists.txt#L233 but `cmake --build` without specified configuration will do Debug build (at least on Windows).

**Review by aneshlya — APPROVED** [review]
Looks good!

**Pull request merged**